collections/guides/use/run-your-own-code-in-a-reactor.md

```diff
@@ -0,0 +1,345 @@
+    state: 'OPEN'
+  };
+
+  const response = await fetch(
+    `https://sandbox.lithic.com/v1/card`,
+    {
+      method: 'post',
+      body: JSON.stringify(body),
+      headers: { 
+        'Authorization': `api-key ${LITHIC_API_KEY}`,
+        'Content-Type': 'application/json',
+      }
+    }
+  );
+  // omitted for brevity: translate response status codes to Basis Theory Reactor errors
+    
+  const lithicCard = await response.json();
+
+  return { 
+    tokenize: {
+      type: 'card',
+      data: {
+        number: lithicCard.pan,
+        expiration_month: lithicCard.exp_month,
+        expiration_year: lithicCard.exp_year,
+        cvc: lithicCard.cvv
+      },
+      metadata: {
+        lithic_token_id: lithicCard.token
+      }
+    }
+  };
+}
+```
+
+Next, we need to create a Private Reactor Formula in Basis Theory containing this `code`, `configuration`, and `request_parameters` by calling the 
+[Create Reactor Formula](https://docs.basistheory.com/#reactor-formulas-create-reactor-formula) API with `type: "private"` (request omitted for brevity). 
+The API will respond with the `id` of our new Private Reactor Formula:
+
+```json
+{
+  "id": "886f14ab-9404-444e-ae44-848c87d1ff88",
+  "type": "private",
+  "status": "verified",
+  "name": "Lithic - Issue Card",
+  "description": "Issues a new card using Lithic",
+  ...
+}
+```
+
+## Using Our Private Reactor Formula
+
+Next, we will use this Reactor Formula to [create a Reactor](https://docs.basistheory.com/#reactors-create-reactor)
+that contains the `LITHIC_API_KEY` to interact on behalf of our company's Lithic account:
+
+```bash
+curl 'https://api.basistheory.com/reactors' \
+  -X 'POST' \
+  -H 'BT-API-KEY: key_PtkNaBwXe23rGt1eZenMMx' \
+  -H 'Content-Type: application/json' \
+  -d '{
+      "name": "Lithic - Issue Card from My Company",
+      "configuration": {
+        "LITHIC_API_KEY": "02f0a915-6226-4a8b-861e-58b08984106f"
+      },
+      "formula": {
+        "id": "886f14ab-9404-444e-ae44-848c87d1ff88"
+      }
+  }'
+```
+
+The Basis Theory API will respond with the `id` of our new Reactor, which we will use later to invoke our Reactor:
+
+```json
+{
+  "id": "ab8a1c9a-8666-449d-91c7-6c0d77bf7d02",
+  "name": "Lithic - Issue Card from My Company",
+  "configuration": {
+    "LITHIC_API_KEY": "02f0a915-6226-4a8b-861e-58b08984106f"
+  },
+  "formula": {...},
+  ...
+}
+```
+
+Now that we have created a Reactor that uses our private formula, we can now make requests to it. 
+We will use the [Invoke a Reactor](https://docs.basistheory.com/#reactors-invoke-a-reactor) endpoint with `args`
+that instruct our Reactor to issue a card called `John's Travel Card` with a total spending limit of $500.
+
+```bash
+curl 'https://api.basistheory.com/reactors/ab8a1c9a-8666-449d-91c7-6c0d77bf7d02/react' \
+  -X 'POST' \
+  -H 'BT-API-KEY: key_PtkNaBwXe23rGt1eZenMMx' \
+  -H 'Content-Type: application/json' \
+  -d '{
+      "args": {
+        "memo": "John'\''s Travel Card",
```

Comment:
does `John's Travel Card` parse correctly? just interesting that we have to escape the single quote with the single quotes around it

Reply:
nvm. looks like it does. I missed this whole thing being wrapped in single quotes. confusing

Reply:
It looks strange, but it works - I copied this out of postman's curl translation, and I tried removing the extra quotes but it becomes an invalid command without them